src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -128,33 +128,57 @@ class EmitFunctionBuilder[F >: Null](
     val f = compareMap.getOrElseUpdate((t, op, missingGreatest), {
       val ti = typeToTypeInfo(t)
       val rt = if (op == CodeOrdering.compare) typeInfo[Int] else typeInfo[Boolean]
-      val newMB = newMethod(Array[TypeInfo[_]](typeInfo[Region], typeInfo[Boolean], ti, typeInfo[Region], typeInfo[Boolean], ti), rt)
-      val ord = t.codeOrdering(newMB)
-      val r1 = newMB.getArg[Region](1)
-      val r2 = newMB.getArg[Region](4)
-      val m1 = newMB.getArg[Boolean](2)
-      val v1 = newMB.getArg(3)(ti)
-      val m2 = newMB.getArg[Boolean](5)
-      val v2 = newMB.getArg(6)(ti)
-      val c: Code[_] = op match {
-        case CodeOrdering.compare => ord.compare(r1, (m1, coerce[ord.T](v1)), r2, (m2, coerce[ord.T](v2)), missingGreatest)
-        case CodeOrdering.equiv => ord.equiv(r1, (m1, coerce[ord.T](v1)), r2, (m2, coerce[ord.T](v2)), missingGreatest)
-        case CodeOrdering.lt => ord.lt(r1, (m1, coerce[ord.T](v1)), r2, (m2, coerce[ord.T](v2)), missingGreatest)
-        case CodeOrdering.lteq => ord.lteq(r1, (m1, coerce[ord.T](v1)), r2, (m2, coerce[ord.T](v2)), missingGreatest)
-        case CodeOrdering.gt => ord.gt(r1, (m1, coerce[ord.T](v1)), r2, (m2, coerce[ord.T](v2)), missingGreatest)
-        case CodeOrdering.gteq => ord.gteq(r1, (m1, coerce[ord.T](v1)), r2, (m2, coerce[ord.T](v2)), missingGreatest)
-        case CodeOrdering.neq => !ord.equiv(r1, (m1, coerce[ord.T](v1)), r2, (m2, coerce[ord.T](v2)), missingGreatest)
+
+      val newMB = if (t.required) {
```

Comment:
is this a bug fix or a change to take advantage of requiredness to do less work?

Reply:
the latter.

Reply:
I should probably just add a flag, though.